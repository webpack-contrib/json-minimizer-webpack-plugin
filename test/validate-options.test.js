import JsonMinimizerPlugin from "../src";

describe("validation", () => {
  it("validation", () => {
    /* eslint-disable no-new */
    expect(() => {
      new JsonMinimizerPlugin({ test: /foo/ });
    }).not.toThrow();

    expect(() => {
      new JsonMinimizerPlugin({ test: "foo" });
    }).not.toThrow();

    expect(() => {
      new JsonMinimizerPlugin({ test: [/foo/] });
    }).not.toThrow();

    expect(() => {
      new JsonMinimizerPlugin({ test: [/foo/, /bar/] });
    }).not.toThrow();

    expect(() => {
      new JsonMinimizerPlugin({ test: ["foo", "bar"] });
    }).not.toThrow();

    expect(() => {
      new JsonMinimizerPlugin({ test: [/foo/, "bar"] });
    }).not.toThrow();

    expect(() => {
      new JsonMinimizerPlugin({ test: true });
    }).toThrowErrorMatchingSnapshot();

    expect(() => {
      new JsonMinimizerPlugin({ test: [true] });
    }).toThrowErrorMatchingSnapshot();

    expect(() => {
      new JsonMinimizerPlugin({ include: /foo/ });
    }).not.toThrow();

    expect(() => {
      new JsonMinimizerPlugin({ include: "foo" });
    }).not.toThrow();

    expect(() => {
      new JsonMinimizerPlugin({ include: [/foo/] });
    }).not.toThrow();

    expect(() => {
      new JsonMinimizerPlugin({ include: [/foo/, /bar/] });
    }).not.toThrow();

    expect(() => {
      new JsonMinimizerPlugin({ include: ["foo", "bar"] });
    }).not.toThrow();

    expect(() => {
      new JsonMinimizerPlugin({ include: [/foo/, "bar"] });
    }).not.toThrow();

    expect(() => {
      new JsonMinimizerPlugin({ include: true });
    }).toThrowErrorMatchingSnapshot();

    expect(() => {
      new JsonMinimizerPlugin({ include: [true] });
    }).toThrowErrorMatchingSnapshot();

    expect(() => {
      new JsonMinimizerPlugin({ exclude: /foo/ });
    }).not.toThrow();

    expect(() => {
      new JsonMinimizerPlugin({ exclude: "foo" });
    }).not.toThrow();

    expect(() => {
      new JsonMinimizerPlugin({ exclude: [/foo/] });
    }).not.toThrow();

    expect(() => {
      new JsonMinimizerPlugin({ exclude: [/foo/, /bar/] });
    }).not.toThrow();

    expect(() => {
      new JsonMinimizerPlugin({ exclude: ["foo", "bar"] });
    }).not.toThrow();

    expect(() => {
      new JsonMinimizerPlugin({ exclude: [/foo/, "bar"] });
    }).not.toThrow();

    expect(() => {
      new JsonMinimizerPlugin({ exclude: true });
    }).toThrowErrorMatchingSnapshot();

    expect(() => {
      new JsonMinimizerPlugin({ exclude: [true] });
    }).toThrowErrorMatchingSnapshot();

    expect(() => {
      new JsonMinimizerPlugin({ minimizerOptions: {} });
    }).not.toThrow();

    expect(() => {
      new JsonMinimizerPlugin({ minimizerOptions: { replacer: ["test", 2] } });
    }).not.toThrow();

    expect(() => {
      new JsonMinimizerPlugin({ minimizerOptions: { replacer: () => {} } });
    }).not.toThrow();

    expect(() => {
      new JsonMinimizerPlugin({ minimizerOptions: { replacer: null } });
    }).not.toThrow();

    expect(() => {
      new JsonMinimizerPlugin({ minimizerOptions: { space: 2 } });
    }).not.toThrow();

    expect(() => {
      new JsonMinimizerPlugin({ minimizerOptions: { space: "  " } });
    }).not.toThrow();

    expect(() => {
      new JsonMinimizerPlugin({ minimizerOptions: { space: null } });
    }).not.toThrow();

    expect(() => {
      new JsonMinimizerPlugin({ minimizerOptions: null });
    }).toThrowErrorMatchingSnapshot();

    expect(() => {
      new JsonMinimizerPlugin({ unknown: true });
    }).toThrowErrorMatchingSnapshot();
    /* eslint-enable no-new */
  });
});
