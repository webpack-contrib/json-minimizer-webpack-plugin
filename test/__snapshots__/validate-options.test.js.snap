// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validation 1`] = `
"Invalid options object. Json Minimizer Plugin has been initialized using an options object that does not match the API schema.
 - options.test should be one of these:
   [RegExp | non-empty string, ...] | RegExp | non-empty string
   -> Filtering rules.
   -> Read more at https://github.com/webpack-contrib/json-minimizer-webpack-plugin#test
   Details:
    * options.test should be an array:
      [RegExp | non-empty string, ...]
    * options.test should be one of these:
      RegExp | non-empty string
      -> Filtering rule as regex or string.
      Details:
       * options.test should be an instance of RegExp.
       * options.test should be a non-empty string."
`;

exports[`validation 2`] = `
"Invalid options object. Json Minimizer Plugin has been initialized using an options object that does not match the API schema.
 - options.test should be one of these:
   [RegExp | non-empty string, ...] | RegExp | non-empty string
   -> Filtering rules.
   -> Read more at https://github.com/webpack-contrib/json-minimizer-webpack-plugin#test
   Details:
    * options.test[0] should be one of these:
      RegExp | non-empty string
      -> Filtering rule as regex or string.
      Details:
       * options.test[0] should be an instance of RegExp.
       * options.test[0] should be a non-empty string."
`;

exports[`validation 3`] = `
"Invalid options object. Json Minimizer Plugin has been initialized using an options object that does not match the API schema.
 - options.include should be one of these:
   [RegExp | non-empty string, ...] | RegExp | non-empty string
   -> Filtering rules.
   -> Read more at https://github.com/webpack-contrib/json-minimizer-webpack-plugin#include
   Details:
    * options.include should be an array:
      [RegExp | non-empty string, ...]
    * options.include should be one of these:
      RegExp | non-empty string
      -> Filtering rule as regex or string.
      Details:
       * options.include should be an instance of RegExp.
       * options.include should be a non-empty string."
`;

exports[`validation 4`] = `
"Invalid options object. Json Minimizer Plugin has been initialized using an options object that does not match the API schema.
 - options.include should be one of these:
   [RegExp | non-empty string, ...] | RegExp | non-empty string
   -> Filtering rules.
   -> Read more at https://github.com/webpack-contrib/json-minimizer-webpack-plugin#include
   Details:
    * options.include[0] should be one of these:
      RegExp | non-empty string
      -> Filtering rule as regex or string.
      Details:
       * options.include[0] should be an instance of RegExp.
       * options.include[0] should be a non-empty string."
`;

exports[`validation 5`] = `
"Invalid options object. Json Minimizer Plugin has been initialized using an options object that does not match the API schema.
 - options.exclude should be one of these:
   [RegExp | non-empty string, ...] | RegExp | non-empty string
   -> Filtering rules.
   -> Read more at https://github.com/webpack-contrib/json-minimizer-webpack-plugin#exclude
   Details:
    * options.exclude should be an array:
      [RegExp | non-empty string, ...]
    * options.exclude should be one of these:
      RegExp | non-empty string
      -> Filtering rule as regex or string.
      Details:
       * options.exclude should be an instance of RegExp.
       * options.exclude should be a non-empty string."
`;

exports[`validation 6`] = `
"Invalid options object. Json Minimizer Plugin has been initialized using an options object that does not match the API schema.
 - options.exclude should be one of these:
   [RegExp | non-empty string, ...] | RegExp | non-empty string
   -> Filtering rules.
   -> Read more at https://github.com/webpack-contrib/json-minimizer-webpack-plugin#exclude
   Details:
    * options.exclude[0] should be one of these:
      RegExp | non-empty string
      -> Filtering rule as regex or string.
      Details:
       * options.exclude[0] should be an instance of RegExp.
       * options.exclude[0] should be a non-empty string."
`;

exports[`validation 7`] = `
"Invalid options object. Json Minimizer Plugin has been initialized using an options object that does not match the API schema.
 - options.minimizerOptions should be an object:
   object { replacer?, space? }
   -> Options for \`jsonMinimizerOptions\`.
   -> Read more at https://github.com/webpack-contrib/json-minimizer-webpack-plugin#minimizeroptions"
`;

exports[`validation 8`] = `
"Invalid options object. Json Minimizer Plugin has been initialized using an options object that does not match the API schema.
 - options has an unknown property 'unknown'. These properties are valid:
   object { test?, include?, exclude?, minimizerOptions? }"
`;
